src/App.jsx

```diff
@@ -1,248 +1,223 @@
-/* eslint-disable jsx-a11y/accessible-emoji */
-import React from 'react';
+import React, { useState } from 'react';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import categoriesFromServer from './api/categories';
-// import productsFromServer from './api/products';
-
-// const products = productsFromServer.map((product) => {
-//   const category = null; // find by product.categoryId
-//   const user = null; // find by category.ownerId
-
-//   return null;
-// });
-
-export const App = () => (
-  <div className="section">
-    <div className="container">
-      <h1 className="title">Product Categories</h1>
-
-      <div className="block">
-        <nav className="panel">
-          <p className="panel-heading">Filters</p>
-
-          <p className="panel-tabs has-text-weight-bold">
-            <a
-              data-cy="FilterAllUsers"
-              href="#/"
-            >
-              All
-            </a>
-
-            <a
-              data-cy="FilterUser"
-              href="#/"
-            >
-              User 1
-            </a>
-
-            <a
-              data-cy="FilterUser"
-              href="#/"
-              className="is-active"
-            >
-              User 2
-            </a>
-
-            <a
-              data-cy="FilterUser"
-              href="#/"
-            >
-              User 3
-            </a>
-          </p>
+import usersFromServer from './api/users';
+import categoriesFromServer from './api/categories';
+import productsFromServer from './api/products';
+
+export const App = () => {
+  const [selectedUser, setSelectedUser] = useState(null);
+  const [searchValue, setSearchValue] = useState('');
+
+  const handleUserFilter = (userId) => {
+    setSelectedUser(userId);
+  };
+
+  const handleSearchInputChange = (search) => {
+    setSearchValue(search.target.value);
+  };
+
+  const clearSearchInput = () => {
+    setSearchValue('');
+  };
+
+  const resetAllFilters = () => {
+    setSelectedUser(null);
+    setSearchValue('');
+  };
+
+  const filteredProducts = productsFromServer.filter((product) => {
+    const { categoryId } = product;
+
+    if (selectedUser !== null
+      && selectedUser !== categoriesFromServer
+        .find(
+          categoryFromServer => categoryFromServer.id === categoryId
+        ).ownerId) {
```

Comment:
This condition looks too complicated. Would be great to simplify it and use variables